pkg/controller/backupinfrastructure/backup_infrastructure_control.go

```diff
@@ -0,0 +1,372 @@
+// implements the documented semantics for BackupInfrastructures. You should use an instance returned from NewDefaultControl() for any
+// scenario other than testing.
+func NewDefaultControl(k8sGardenClient kubernetes.Client, k8sGardenInformers gardeninformers.Interface, secrets map[string]*corev1.Secret, imageVector imagevector.ImageVector, identity *gardenv1beta1.Gardener, config *componentconfig.ControllerManagerConfiguration, recorder record.EventRecorder, updater UpdaterInterface) ControlInterface {
+	return &defaultControl{k8sGardenClient, k8sGardenInformers, secrets, imageVector, identity, config, recorder, updater}
+}
+
+type defaultControl struct {
+	k8sGardenClient    kubernetes.Client
+	k8sGardenInformers gardeninformers.Interface
+	secrets            map[string]*corev1.Secret
+	imageVector        imagevector.ImageVector
+	identity           *gardenv1beta1.Gardener
+	config             *componentconfig.ControllerManagerConfiguration
+	recorder           record.EventRecorder
+	updater            UpdaterInterface
+}
+
+func (c *defaultControl) ReconcileBackupInfrastructure(obj *gardenv1beta1.BackupInfrastructure, key string) (bool, error) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		return true, err
+	}
+
+	var (
+		backupInfrastructure       = obj.DeepCopy()
+		operationID                = utils.GenerateRandomString(8)
+		backupInfrastructureLogger = logger.NewBackupInfrastructureLogger(logger.Logger, backupInfrastructure.Name, backupInfrastructure.Namespace, operationID)
+		lastOperation              = backupInfrastructure.Status.LastOperation
+		operationType              = computeOperationType(lastOperation)
+	)
+
+	logger.Logger.Infof("[BACKUPINFRASTRUCTURE RECONCILE] %s", key)
+	backupInfrastructureJSON, _ := json.Marshal(backupInfrastructure)
+	backupInfrastructureLogger.Debugf(string(backupInfrastructureJSON))
+
+	op, err := operation.NewWithBackupInfrastructure(backupInfrastructure, backupInfrastructureLogger, c.k8sGardenClient, c.k8sGardenInformers, c.identity, c.secrets, c.imageVector)
+	if err != nil {
+		backupInfrastructureLogger.Errorf("Could not initialize a new operation: %s", err.Error())
+		return true, err
+	}
+
+	// The deletionTimestamp labels a BackupInfrastructure as intended to get deleted. Before deletion,
+	// it has to be ensured that no Infrastructure resources are depending on the BackupInfrastructure anymore.
+	// When this happens the controller will remove the finalizer from the BackupInfrastructure so that it can be garbage collected.
+	if backupInfrastructure.DeletionTimestamp != nil {
+
+		gracePeriod := time.Hour * 24 * time.Duration(*backupInfrastructure.Spec.DeletionGracePeriodDays)
+		if time.Now().Sub(backupInfrastructure.DeletionTimestamp.Time) > gracePeriod {
+			c.recorder.Eventf(backupInfrastructure, corev1.EventTypeNormal, gardenv1beta1.EventDeleting, "[%s] Deleting Backup Infrastructure", operationID)
```

Comment:
Didn't we say that we only want to send events in case of errors?

Reply:
Yes, sorry. This should have been removed.